app/gui2/src/util/ast/reactive.ts

```diff
@@ -0,0 +1,66 @@
+import type { AstId } from 'shared/ast'
+import { type FixedMap, MutableModule } from 'shared/ast'
+import { customRef, markRaw } from 'vue'
+import * as Y from 'yjs'
+
+const WrappedValue = Symbol('wrapped')
+
+export function reactiveModule(doc: Y.Doc): {
+  module: MutableModule
+  triggerAst: (id: AstId) => void
+} {
+  const triggers = new WeakMap<object, () => void>()
+  const tracks = new WeakMap<object, () => void>()
+  const wrap = <Fields>(wrapped: FixedMap<Fields>) => {
+    if (Object.hasOwnProperty.bind(wrapped)(WrappedValue)) return wrapped
+    const track = () => {
+      const track = tracks.get(wrapped)
+      if (track) {
+        track()
+      } else {
+        const { track, trigger } = customRef((track, trigger) => ({
+          get: () => ({ track, trigger }),
+          set: () => {},
+        })).value
+        triggers.set(wrapped, trigger)
+        tracks.set(wrapped, track)
+        track()
+      }
+    }
+    return Object.defineProperty(
```

Comment:
This whole `defineProperty` call looks a bit scuffed. First we are creating the object with functions, then we are adding a `WrappedValue` key to it through `defineProperty`. On top of that, the property descriptor is created with `Object.assign(Object.create(null),...)`. Why?

Could this be simplified to simple object literal?
```ts
return {
	[WrappedValue]: wrapped,
	get: ...
}
```